x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -504,6 +622,7 @@ private static RoleDescriptor.IndicesPrivileges parseIndex(String roleName, XCon
         String[] grantedFields = null;
         String[] deniedFields = null;
         boolean allowRestrictedIndices = false;
+        String[] remoteClusters = null;
```

Comment:
It needs to be nullable here because this method is used by both `parseIndex` and `parseRemoteIndex`.  For `parseIndex` the value _should_ be null. It can't pass `null` to the builder -- inside `parseRemoteIndex` we throw if the value is null, and inside `parseIndex` we assert that the value is null (because it should be).